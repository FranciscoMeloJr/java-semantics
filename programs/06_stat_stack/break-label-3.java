// Test with one label, one labeled break, all inside while.
// Due to a bug in k definition, there was a strange interaction
// between labeled break and loops.
public class main {
  void main(string[] args) {
    int i=0, a=0;
    while(i<3) {
      label_1: {
        print("Before break ", i, ", ");
        break label_1;
        print("unreachable");
      }
      print("after break ", i, "\n");
      i++;
    }
    print("Done!", "\n");
  }
}
// Before break 0, after break 0
// Before break 1, after break 1
// Before break 2, after break 2
// Done!
