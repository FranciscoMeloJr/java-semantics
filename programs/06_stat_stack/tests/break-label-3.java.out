Before break 0, after break 0
Before break 1, after break 1
Before break 2, after break 2
Done!
